core/src/main/kotlin/org/evomaster/core/problem/rest/classifier/deterministic/Deterministic400EndpointModel.kt

```diff
@@ -101,20 +102,29 @@ class Deterministic400EndpointModel(
         return AIResponseClassification(probabilities, failedConstraintFields)
     }
 
-    override fun estimateAccuracy(endpoint: Endpoint): Double {
+    /** Default metrics estimates */
+    override fun estimateMetrics(endpoint: Endpoint): ModelEvaluation {
         verifyEndpoint(endpoint)
+        if (!initialized) {
+            // hasn’t learned anything yet → return defaults
+            return ModelEvaluation.DEFAULT_NO_DATA
+        }
 
-        return estimateOverallAccuracy()
+        return ModelEvaluation(
```

Comment:
i don't understand this change. why removing the call to `estimateOverallAccuracy()`?